scripts/core/editor3/components/DraftSpellchecker.jsx

```diff
@@ -0,0 +1,240 @@
+    fullSpellcheck(editorState) {
+        var selectionState = editorState.getSelection();
+        var contentState = editorState.getCurrentContent();
+        var blocks = contentState.getBlocksAsArray();
+
+        return blocks.reduce(
+            (crtContentState, block) =>
+                this.blockSpellcheck(block, crtContentState, selectionState).then(
+                    (newContentState) => this.restoreSelection(
+                        editorState, contentState, newContentState, selectionState)
+                )
+            , contentState);
+    }
+
+    /** Spellcheck the text from current editor block and return a new editor state */
+    currentBlockSpellcheck(editorState) {
+        var selectionState = editorState.getSelection();
+        var contentState = editorState.getCurrentContent();
+        var block = contentState.getBlockForKey(selectionState.getStartKey());
+
+        return this.blockSpellcheck(block, contentState, selectionState).then(
+            (newContentState) => this.restoreSelection(editorState, contentState, newContentState, selectionState)
+        );
+    }
+
+    /** Show the contextual menu at the given position */
+    showContextMenu(text, textSelection, leftPosition, topPosition) {
+        this.leftPosition = leftPosition;
+        this.topPosition = topPosition;
+        this.textSelection = textSelection;
+
+        this.wordSelection = this.getCrtWordSelection(text, textSelection.anchorOffset);
+        if (!this.wordSelection) {
+            this.show = false;
+            return;
+        }
+
+        this.initSuggestions();
+    }
+
+    /** Closes the context menu and calls editor on close context menu actions */
+    closeContextMenu() {
+        if (this.show) {
+            this.show = false;
+            if (this.onCloseContextMenu) {
+                this.onCloseContextMenu();
+            }
+        }
+    }
+
+    /** Extract the word selection at the index position */
+    getCrtWordSelection(text, index) {
+        const regexp = /[0-9a-zA-Z\u00C0-\u1FFF\u2C00-\uD7FF]+/g;
+        var match;
+
+        while (!_.isNil(match = regexp.exec(text))) {
+            var word = match[0];
+
+            if (match.index <= index && index <= match.index + word.length) {
+                return {
+                    word: word,
+                    index: match.index
+                };
+            }
+        }
+
+        return null;
+    }
+
+    /** Fetch the suggestions for the current word */
+    initSuggestions() {
+        this.show = false;
+        if (this.getSuggestions) {
+            this.getSuggestions(this.wordSelection.word).then((suggestions) => {
+                this.suggestions = suggestions;
+                this.show = true;
+                if (this.onOpenContextMenu) {
+                    this.onOpenContextMenu();
+                }
+            });
+        } else {
+            this.suggestions = [];
```

Comment:
Lines 187 to 191 are the same as 180 to 184 so I would create a function here:
```js
const show = (s) => {
    this.suggestions = s;
    this.show = true;

    if (this.onOpenContextMenu) {
        this.onOpenContextMenu();
    }
};
```
and then use that:
```js
if (this.getSuggestions) {
    this.getSuggestions(this.wordSelection.word).then(show);
} else {
    show([]);
}
```
to avoid repetition...